etcdserver/server.go

```diff
@@ -466,11 +470,24 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 // modify a server's fields after it has been sent to Start.
 // It also starts a goroutine to publish its server information.
 func (s *EtcdServer) Start() {
+	s.wg.Add(4)
```

Comment:
Sure. The current patch is a little messy-- too many `wg.Add`'s / `wg.Done`'s flying around. Going to wrap it all in something like `func (s* EtcdServer) attachGoRoutine(f func())`.